torch/csrc/jit/passes/quantization.cpp

```diff
@@ -695,6 +744,36 @@ void InsertObserversHelper::addIntermediateValuesToSkipObserver(
   }
 }
 
+void InsertObserversHelper::fillBoundaryValueMap(
+    script::Module& module, const std::string& method_name) {
+  auto graph = module.get_method(method_name).graph();
+  std::stack<Block*> blocks_to_visit;
+  blocks_to_visit.push(graph->block());
+  auto* self = graph->inputs()[0];
+  while (!blocks_to_visit.empty()) {
+    Block* b = blocks_to_visit.top();
+    blocks_to_visit.pop();
+    for (Node* n : b->nodes()) {
+      if (n->kind() == prim::CallMethod) {
+        auto m = getInvokedModule(module, n, self);
+        auto g = m.get_method(n->s(attr::name)).graph();
+        // add mapping from callsite value to value in called graph
+        for (auto i = 0; i < g->outputs().size(); ++i) {
+          auto* return_val = g->outputs()[i];
+          boundary_value_map_[n->outputs()[i]].push_back(return_val);
+        }
+        for (auto i = 0; i < g->inputs().size(); ++i) {
+          auto* input_val = g->inputs()[i];
+          boundary_value_map_[n->inputs()[i]].push_back(input_val);
+        }
+      }
+      for (Block* subblock : n->blocks()) {
+        blocks_to_visit.push(subblock);
+      }
+    }
+  }
+}
+
```

Comment:
Suggest refactoring pre-process to have graph cleanup steps (only replaceConv and FuseLinear). The rest of the observer specific steps (addIntermediate etc) can be in a different function.

Reply:
Sure, I'll refactor them later

Reply:
Please file a task for this purpose